src/gapvector.h

```diff
@@ -192,6 +183,18 @@ class Gapvector {
         bufferEnd = gapEnd;
     }
 
+    template <std::ranges::input_range InputRng>
```

Comment:
I’m not familiar with ranges too much, how does this work? Does this differ from the iterator constructor?